solver/src/naive_solver/multi_order_solver.rs

```diff
@@ -160,6 +195,54 @@ fn compute_uniswap_in(out: U256, shortage: &TokenContext, excess: &TokenContext)
     U256::from(1000) * out * excess.reserve / (U256::from(997) * (shortage.reserve - out)) + 1
 }
 
+///
+/// Returns true if for each trade the executed price is not smaller than the limit price
+/// Thus we ensure that `buy_token_price / sell_token_price >= limit_buy_amount / limit_sell_amount`
+///
+fn is_valid_solution(solution: &SinglePairSettlement) -> bool {
+    for trade in solution.trades.iter() {
+        let order = trade.order;
+        let buy_token_price = match solution.clearing_prices.get(&order.buy_token) {
+            Some(price) => price,
+            None => return false,
+        };
+        let sell_token_price = match solution.clearing_prices.get(&order.sell_token) {
+            Some(price) => price,
+            None => return false,
+        };
+
+        if order.sell_amount * buy_token_price < order.buy_amount * sell_token_price {
+            return false;
+        }
```

Comment:
In the smart contract this is
```rust
assert!(sell_amount * sell_price >= buy_amount * buy_price);
```
which seems different.

Reply:
Correct, the prices are inverted here.

The example I like to use to get the order right is:
> Beer costs 4€ and chocolate 1.50€
> So 3 beers are equivalent to 8 chocolates because and `3 * 4€ = 8 * 1.50€`

Reply:
Really good catch! Will adjust the unit test to also catch this.